interpreter/cling/lib/Interpreter/NullDerefProtectionTransformer.cpp

```diff
@@ -175,17 +193,10 @@ namespace cling {
 
   ASTTransformer::Result
   NullDerefProtectionTransformer::Transform(clang::Decl* D) {
-    FunctionDecl* FD = dyn_cast<FunctionDecl>(D);
-    if (!FD || FD->isFromASTFile())
-      return Result(D, true);
-
-    CompoundStmt* CS = dyn_cast_or_null<CompoundStmt>(FD->getBody());
-    if (!CS)
-      return Result(D, true);
 
     PointerCheckInjector injector(*m_Sema);
-    injector.Visit(CS);
+    injector.VisitDecl(D);
```

Comment:
Now it builds all good, but it does not do the check, because VisitDecl is just this:
clang::RecursiveASTVisitor<cling::PointerCheckInjector>::VisitDecl (this=0x7fffffffa660, D=0x16a7bd0)
    at /home/biancacr/work2/root/interpreter/cling/../llvm/src/tools/clang/include/clang/AST/RecursiveASTVisitor.h:403
403	  bool VisitDecl(Decl *D) { return true; }

Should I give an explicit definition?, but then that has to use either VisitStmt and then we miss our cases where we have decls inside decls..